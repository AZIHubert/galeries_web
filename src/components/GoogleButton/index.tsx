import * as React from 'react';

interface GoogleButtonI {
  loading: boolean;
  setLoading: React.Dispatch<React.SetStateAction<boolean>>;
  type: 'login' | 'signin';
}

const GoogleButton = ({
  loading,
  setLoading,
  type,
}: GoogleButtonI) => (
  <button
    data-testid='button'
    disabled={loading}
    onClick={() => setLoading(true)}
  >
    {type === 'signin' && 'sign in '}
    {type === 'login' && 'log in '}
    with Google
  </button>
);

export default GoogleButton;
