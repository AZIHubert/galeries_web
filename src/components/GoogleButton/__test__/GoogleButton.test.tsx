import * as React from 'react';
import renderer from 'react-test-renderer';
import {
  fireEvent,
  render,
  screen,
} from '@testing-library/react';

import GoogleButton from '../index';

const Container = () => {
  const [loading, setLoading] = React.useState(false);
  return (
    <GoogleButton
      loading={loading}
      setLoading={setLoading}
      type='login'
    />
  );
};

describe('GoogleButton', () => {
  const setLoading = jest.fn;
  it('renders without crashing', () => {
    const tree = renderer.create(
      <GoogleButton
        loading={false}
        setLoading={setLoading}
        type={'login'}
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should display \'log in with Google if type is \'login\'', () => {
    const { getByTestId } = render(
      <GoogleButton
        loading={false}
        setLoading={setLoading}
        type='login'
      />,
    );
    const button = getByTestId('button');
    expect(button).toHaveTextContent('log in with Google');
  });
  it('should display \'sign in with facebook if type is \'signin\'', () => {
    const { getByTestId } = render(
      <GoogleButton
        loading={false}
        setLoading={setLoading}
        type='signin'
      />,
    );
    const button = getByTestId('button');
    expect(button).toHaveTextContent('sign in with Google');
  });
  it('should not be disable on mount', () => {
    const { getByTestId } = render(
      <GoogleButton
        loading={false}
        setLoading={setLoading}
        type='signin'
      />,
    );
    const button = getByTestId('button');
    expect(button).not.toBeDisabled();
  });
  it('should be disabled after clicking', () => {
    const { getByTestId } = render(<Container />);
    const button = getByTestId('button');
    fireEvent.click(button);
    const buttonAfterClick = screen.getByTestId('button');
    expect(buttonAfterClick).toBeDisabled();
  });
});
