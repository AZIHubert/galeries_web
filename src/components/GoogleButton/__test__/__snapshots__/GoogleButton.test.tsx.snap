// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GoogleButton renders without crashing 1`] = `
<button
  data-testid="button"
  disabled={false}
  onClick={[Function]}
>
  log in 
  with Google
</button>
`;
